src/test/java/uk/gov/pay/api/resources/directdebit/DirectDebitPaymentsResourceIT.java

```diff
@@ -0,0 +1,181 @@
+                .body("reference", is(REFERENCE))
+                .body("state.status", is(status))
+                .body("state.finished", is(finished))
+                .body("_links.events.href", is(paymentEventsLocationFor(PAYMENT_ID)))
+                .body("_links.events.method", is("GET"))
+                .body("_links.self.href", is(paymentLocationFor(PAYMENT_ID)))
+                .body("_links.self.method", is("GET"))
+                .body("_links.mandate.href", is(mandateLocationFor(MANDATE_ID)))
+                .body("_links.mandate.method", is("GET"));
+    }
+
+    @Test
+    public void createPayment_successWhenDescriptionNotPresent() throws JsonProcessingException {
+        final Account account = new Account(GATEWAY_ACCOUNT_ID, TokenPaymentType.DIRECT_DEBIT);
+        final String status = "created";
+        final boolean finished = false;
+
+        DirectDebitConnectorCreatePaymentResponse connectorResponse = aDirectDebitConnectorCreatePaymentResponse()
+                .withPaymentExternalId(PAYMENT_ID)
+                .withAmount(AMOUNT)
+                .withPaymentProvider(PAYMENT_PROVIDER)
+                .withCreatedDate(CREATED_DATE)
+                .withMandateId(MANDATE_ID)
+                .withProviderId(PROVIDER_ID)
+                .withState(new DirectDebitPaymentState(status, finished, null))
+                .withReference(REFERENCE)
+                .build();
+
+        connectorDDMockClient.respondWithPaymentCreated(connectorResponse, account.getAccountId());
+
+        String request = new Gson().toJson(Map.of(
+                "amount", AMOUNT,
+                "reference", REFERENCE,
+                "mandate_id", MANDATE_ID));
+
+        postPaymentResponse(request)
+                .statusCode(201)
+                .body("$", not(hasKey("description")));
+    }
+
+    @Test
+    @Parameters(method = "parametersForValidation")
+    public void validationFailures(
+            @Nullable Object amount,
+            @Nullable Object reference,
+            @Nullable Object description,
+            @Nullable Object mandateId,
+            String expectedErrorMessage,
+            String field,
+            String errorCode) {
+
+        var payload = new HashMap<>();
+        Optional.ofNullable(amount).ifPresent(ref -> payload.put("amount", ref));
+        Optional.ofNullable(reference).ifPresent(ref -> payload.put("reference", ref));
+        Optional.ofNullable(description).ifPresent(ref -> payload.put("description", ref));
+        Optional.ofNullable(mandateId).ifPresent(ref -> payload.put("mandate_id", ref));
+
+        postPaymentResponse(new Gson().toJson(payload))
+                .statusCode(422)
+                .contentType(JSON)
+                .body("code", is(errorCode))
+                .body("field", is(field))
+                .body("description", is(expectedErrorMessage));
+    }
+
+    private Object[] parametersForValidation() {
```

Comment:
Just a passing comment before I go for lunch, happy to carry on when I'm back. I find this a little difficult to read at first. Might it be nice to create an inner class to represent these parameters and then return an array of them, you could even have a `toPayload` method on such class which will simplify the test method itself...e.g `var payload = parameters.toPayload()` and `.body("description", is(parameters.getExpectedMessage())` the test method itself becomes something like 
`public void validationFailures(TestParameters parameters)`

Happy to explain is that doesn't make sense :)